src/sentry/api/bases/user.py

```diff
@@ -26,8 +27,16 @@ def has_object_permission(self, request: Request, view, user: User | RpcUser | N
             return True
         if request.auth:
             return False
-        if is_active_superuser(request):
-            return True
+
+        # populate request.access for SaaS superuser
+        SUPERUSER_ORG_ID = getattr(settings, "SUPERUSER_ORG_ID", None)
+        if SUPERUSER_ORG_ID and is_active_superuser(request):
+            org = organization_service.get_organization_by_id(id=SUPERUSER_ORG_ID)
+            self.determine_access(request, org)
```

Comment:
this permission class overwrites the `has_object_permission` function from its parent class which normally calls `self.determine_access` and populates the request.access object. in order to know if a superuser has write permission we have to call this function